source/brailleInput.py

```diff
@@ -134,48 +152,64 @@ def _translate(self, endWord: bool) -> bool:
 		pos = self.untranslatedStart + self.untranslatedCursorPos
 		data = u"".join([chr(cell | LOUIS_DOTS_IO_START) for cell in self.bufferBraille[:pos]])
 		mode = louis.dotsIO | louis.noUndefinedDots
-		if (not self.currentFocusIsTextObj or self.currentModifiers) and self._table.contracted:
+		if self.inputKeysUsingEmulation and self._table.contracted:
 			mode |=  louis.partialTrans
 		self.bufferText = louis.backTranslate(
 			[os.path.join(brailleTables.TABLES_DIR, self._table.fileName),
 			"braille-patterns.cti"],
 			data, mode=mode)[0]
 		newText = self.bufferText[oldTextLen:]
+		if _isDebugForBrailleInput():
+			log.debug(f"bufferText after translation = {self.bufferText}, newText = {newText}")
 		if newText:
 			# New text was generated by the cells just entered.
 			if self.useContractedForCurrentFocus or self.currentModifiers:
 				# For contracted braille, an entire word is sent at once.
 				# Don't speak characters as this is sent.
 				# Also, suppress typed characters when emulating a command gesture.
+				if _isDebugForBrailleInput():
+					log.debug("Suppressing typed characters for contracted input")
 				speech._suppressSpeakTypedCharacters(len(newText))
 			else:
 				self._uncontSentTime = time.time()
 			self.untranslatedStart = pos
 			self.untranslatedCursorPos = 0
-			if self.currentModifiers or not self.currentFocusIsTextObj:
+			if _isDebugForBrailleInput():
+				log.debug(
+					f"currentModifiers = {self.currentModifiers}, currentFocusIsTextObj = {self.currentFocusIsTextObj}"
+				)
+			if self.inputKeysUsingEmulation:
 				if len(newText)>1:
 					# Emulation of multiple characters at once is unsupported
 					# Clear newText, so this function returns C{False} if not at end of word
+					if _isDebugForBrailleInput():
+						log.debug(f"Clearing newText, was {newText}")
 					newText = u""
 				else:
 					self.emulateKey(newText)
 			else:
 				self.sendChars(newText)
 
-		if endWord or (newText and (not self.currentFocusIsTextObj or self.currentModifiers)):
+		if endWord or (newText and (self.currentModifiers or not self.useContractedForCurrentFocus)):
```

Comment:
In the old situation, the buffer wasn't cleared when in a text field. In the new situation, the buffer will be cleared when contraction should not apply, either when the table is not contracted or when modifiers are pressed.